src/Util.php

```diff
@@ -24,4 +59,26 @@ public static function getDatasetId($dataset): string
         }
         return "{$id}";
     }
+
+    /**
+     * Sort an array in place by key.
+     *
+     * Recursively applies ksort() to any value in the array that is an array.
+     *
+     * @param $array
+     *   The array to be sorted.
+     * @param $flags
+     *   Flags to pass along to ksort().
+     *
+     * @see \ksort()
+     */
+    public static function recursiveKeySort(&$array, int $flags = SORT_REGULAR): void
+    {
+        foreach ($array as &$value) {
+            if (is_array($value)) {
+                static::recursiveKeySort($value);
+            }
+        }
+        ksort($array, $flags);
+    }
```

Comment:
@paul-m just want to check, this should have no effect on JSON arrays, correct? I would think if I pass it `['c', 'b', 'a']` it will still see the "keys" (thanks PHP!) as 0, 1, 2 and leave the order in tact?

Reply:
Correct.